### DISCUSSION TITLE
onClick method inside drop down menu option doesn't respond when only one option is shown to the user

### DISCUSSION THREAD

**Discussion by nora-kauczor** [neuefische/questions#836] [Web]
- I want to show different versions of a drop down menu with either one or two options (see JSX below)
- Either, see user sees what he typed + a suggestion for what he might be looking for (two options in the drop down), or, if there's no suggestion, he only sees what he typed (one option in the drop down)
- The options receive an onClick event that calls a function which renders the clicked text below
- For now, the onClick is working when two options are shown, but not, when only one is shown. In that case, when clicking on the displayed option, nothing happens (no error is shown either)
- You can find the file [here](https://github.com/Larissartoricader/Capstone/blob/feature/refine-form-2/components/RecipeForm.js), it's a little messy for now for there's other parts to adjust after this.
- I've copied the relevant parts here: 

```
const DropDownOption = styled.option`
  width: 80vw;
`;

const [ingredientSuggestion, setIngredientSuggestion] = useState();

// selectedIngredients are the strings that will be rendered (displayed below the input field) when having been clicked by the user:
const [selectedIngredients, setSelectedIngredients] = useState([]);

  function selectIngredient(event) {
    if (selectedIngredients.includes(event.target.value)) {
      setAlreadySelectedIngredient(true);
    } else {
      setSelectedIngredients([...selectedIngredients, event.target.value]);
      setIngredientsInput("");
      setIngredientSuggestion("");
    }
  }
```

Inside the JSX:
```
{ingredientsInput && (
          <select size={ingredientSuggestion ? "2" : "1"}>
            {ingredientSuggestion && (
              <DropDownOption onClick={selectIngredient}>
                {ingredientSuggestion}
              </DropDownOption>
            )}
            {ingredientsInput && (
              <DropDownOption onClick={selectIngredient}>
                {ingredientsInput}
              </DropDownOption>
            )}
          </select>
        )}
```

**Comment by F-Kirchhoff** [comment]
The Problem could be resolved by switching select and option for a div and buttons.
